tensorbay/opendataset/PlantSeedlings/loader.py

```diff
@@ -0,0 +1,175 @@
+    (53, 56): "Loose Silky-bent",
+}
+
+
+def PlantSeedlings(path: str) -> Dataset:
+    """Dataloader of PlantSeedlings dataset.
+
+    .. PlantSeedlings: https://vision.eng.au.dk/plant-seedlings-dataset/
+
+    The file structure should be like::
+
+        <path>
+            Fullimages/
+                1/
+                    IMG_0573_12-03.png
+                    ...
+                2/
+                    ...
+                ...
+                SpeciesIndex.pdf
+                SpeciesIndex.xlsx
+            Black-grass/
+                1.png
+                2.png
+                ...
+            Charlock/
+            Cleavers/
+            Common Chickweed/
+            Common wheat/
+            Fat Hen/
+            Loose Silky-bent/
+            Maize/
+            Scentless Mayweed/
+            Shepherd’s Purse/
+            Small-flowered Cranesbill/
+            Sugar beet/
+            Segmented/
+                Black-grass/
+                 1.png
+                 2.png
+                 ...
+                Charlock/
+                Cleavers/
+                Common Chickweed/
+                Common wheat/
+                Fat Hen/
+                Loose Silky-bent/
+                Maize/
+                Scentless Mayweed/
+                Shepherd’s Purse/
+                Small-flowered Cranesbill/
+                Sugar beet/
+
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    for subset in os.listdir(root_path):
+        if subset == "Segmented":
+            continue
+        segment = (
+            _get_segment(root_path, subset)
+            if subset != "Fullimages"
+            else _get_fullimages(root_path)
+        )
+        dataset.add_segment(segment)
+    return dataset
+
+
+def _get_fullimages(root_path: str) -> Segment:
+    segment = Segment("Fullimages")
+    files_path = os.path.join(root_path, "Fullimages")
+    category_map = _get_category_map()
+    for file_path in glob(os.path.join(files_path, "[0-9]*")):
+        file = os.path.basename(file_path)
+        for data in _generate_fullimages_data(os.path.join(files_path, file), category_map[file]):
+            segment.append(data)
+    return segment
+
+
+def _get_category_map() -> Dict[str, str]:
+    category_map = {}
+    for key, value in _FULLIMAGES_MAP.items():
+        for index in range(key[0], key[1] + 1):
+            category_map[str(index)] = value
+    return category_map
+
+
+def _generate_fullimages_data(data_path: str, category: str) -> Iterator[Data]:
+    for data_subset in os.listdir(data_path):
+        data = Data(
+            os.path.join(data_path, data_subset), target_remote_path=f"{category}_{data_subset}"
```

Comment:
Are there duplicate images in different folders?

Reply:
No, Madam.